build.md

```diff
@@ -155,6 +155,35 @@ args:
   - GIT_COMMIT
 ```
 
+### ssh
+
+`ssh` defines an SSH authentication that the image builder SHOULD use during image build (e.g., cloning private repository)
+
+`ssh` property syntax can be either:
+* `default` - let the builder connect to the ssh-agent.
+* `ID=socket` - a key/value definition of an ID and the associated socket
+* `ssh` without any value which is a shortcut of the `default` configuration
+
+`ssh` without any parameter
+```yaml
+build:
+  context: .
+  ssh:   # mount the default ssh agent
+```
+
+Simple`default` sample
+```yaml
+build:
+  context: .
+  ssh: default   # mount the default ssh agent
+```
+Using a custom id `myproject` which will be use in the Dockerfile (i.e., `RUN --mount=type=ssh,id=myproject git clone ...`)
```

Comment:
example uses buildkit extended syntax: https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypessh
I'm fine we have implementation-specific examples, but should be noted as such.

I suggest compose.yaml example is followed by an example Dockerfile using buildkit syntax (with link) as an explicit implementation-specific illustration example.